data-access-layer/eventmesh/ApiServerClient.js

```diff
@@ -142,19 +148,20 @@ class ApiServerClient {
       .then(() => {
         const stream = apiserver
           .apis[resourceGroup][CONST.APISERVER.API_VERSION]
-          .watch.namespaces(CONST.APISERVER.NAMESPACE)[resourceType].getStream({
+          .watch[resourceType].getStream({
             qs: {
-              labelSelector: queryString ? queryString : ''
+              labelSelector: queryString ? queryString : '',
+              timeoutSeconds: CONST.APISERVER.WATCH_TIMEOUT
             }
           });
-        const jsonStream = new JSONStream();
-        stream.pipe(jsonStream);
-        jsonStream.on('data', callback);
-        jsonStream.on('error', err => {
-          logger.error('Error occured during watching', err);
+        stream.on('error', err => {
+          logger.error(`Error occured during watching for resource ${resourceGroup}, ${resourceType}`, err);
           this.registerWatcher(resourceGroup, resourceType, callback, queryString);
           //throw err;
         });
+        const jsonStream = new JSONStream();
+        stream.pipe(jsonStream);
+        jsonStream.on('data', callback);
```

Comment:
How do we want to capture errors on the stream?

Reply:
We are reregistering on error. @subhankarc  Should we do stream.abort() before reregistering the watch on error?

Reply:
Yes makes sense I guess.